CHANGELOG.md

```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## 3.6.0 - 2024-11-18
+- Address Python 3.12+ compatibility issues.
+- Address pydantic v2 compatibility issues.
+- Raise minimum required Python version to 3.8.
+- Raise minimum required `pydantic` version to 2.0.
+
```

Comment:
Should we mention shrub, pytest and flake8 version upgrades here as well? Other than this, LGTM.

Reply:
I omitted pytest and flake8 due to being dev dependencies (not observed by downstream users of the shrub.py package). I can add them nevertheless if you prefer.